main.go

```diff
@@ -94,68 +149,67 @@ type listOutputJSON struct {
 func outputList() {
 	list, err := enumerator.GetDetailedPortsList()
 	if err != nil {
-		outputError(err)
+		output(&genericMessageJSON{
+			EventType: "list",
+			Error:     true,
+			Message:   err.Error(),
+		})
 		return
 	}
 	portsJSON := []*boardPortJSON{}
 	for _, port := range list {
 		portJSON := newBoardPortJSON(port)
 		portsJSON = append(portsJSON, portJSON)
 	}
-	d, err := json.MarshalIndent(&listOutputJSON{
+	output(&listOutputJSON{
 		EventType: "list",
 		Ports:     portsJSON,
-	}, "", "  ")
-	if err != nil {
-		outputError(err)
-		return
-	}
-	syncronizedPrintLn(string(d))
+	})
 }
 
 func newBoardPortJSON(port *enumerator.PortDetails) *boardPortJSON {
 	prefs := properties.NewMap()
-	identificationPrefs := properties.NewMap()
 	portJSON := &boardPortJSON{
-		Address:             port.Name,
-		Label:               port.Name,
-		Protocol:            "serial",
-		ProtocolLabel:       "Serial Port",
-		Prefs:               prefs,
-		IdentificationPrefs: identificationPrefs,
+		Address:       port.Name,
+		Label:         port.Name,
+		Protocol:      "serial",
+		ProtocolLabel: "Serial Port",
+		Properties:    prefs,
 	}
 	if port.IsUSB {
 		portJSON.ProtocolLabel = "Serial Port (USB)"
-		portJSON.Prefs.Set("vendorId", "0x"+port.VID)
-		portJSON.Prefs.Set("productId", "0x"+port.PID)
-		portJSON.Prefs.Set("serialNumber", port.SerialNumber)
-		portJSON.IdentificationPrefs.Set("pid", "0x"+port.PID)
-		portJSON.IdentificationPrefs.Set("vid", "0x"+port.VID)
+		portJSON.Properties.Set("vid", "0x"+port.VID)
+		portJSON.Properties.Set("pid", "0x"+port.PID)
+		portJSON.Properties.Set("serialNumber", port.SerialNumber)
 	}
 	return portJSON
 }
 
-type messageOutputJSON struct {
+type helloMessageJSON struct {
+	EventType       string `json:"eventType"`
+	ProtocolVersion string `json:"protocolVersion"`
```

Comment:
Why not an `int`?

Reply:
mmmhhhhh... I need to change the specification, but you're right, better to be `int`